corehq/apps/es/migration_operations.py

```diff
@@ -95,8 +104,72 @@ def run(self, *args, **kw):
             self.settings_key,
             self.comment,
         ))
+        if self.creation_checks:
+            self._wait_for_primary_shards_to_be_assigned()
+
         manager.index_configure_for_standard_ops(self.name)
 
+    def _validate_disk_under_watermark(self):
+        """
+        Validate disk usage in each data node is under watermarks before creating index
+        Raises Exception and fails if no data node has available disk space
+        """
+        settings = manager.cluster_get_settings()['transient']
+        low_watermark = settings.get("cluster.routing.allocation.disk.watermark.low", "85%")
+
+        fs_info = manager.get_node_fs_stats()
+        has_one_node_with_available_space = False
+
+        def is_data_node(stats):
+            return 'data' in stats['roles']
+
+        def within_low_watermark(stats, low_watermark_percentage):
+            return stats['disk_usage_percentage'] < low_watermark_percentage
+
+        for info in fs_info:
+            total_disk_size = info['total_disk_size']
+            low_watermark_percentage = manager._parse_watermark_to_percentage(total_disk_size, low_watermark)
+
+            if is_data_node(info) and within_low_watermark(info, low_watermark_percentage):
+                has_one_node_with_available_space = True
+            elif not within_low_watermark(info, low_watermark_percentage):
+                log.warning(
+                    f"{info['node_name']} exceeds low disk watermark of {low_watermark_percentage}%. "
+                    f"Available disk space: {100-info['disk_usage_percentage']}%.\n"
+                )
+
+        if not has_one_node_with_available_space:
+            raise Exception("""All data nodes are above low watermark capacity.
```

Comment:
Is there a better class of `Exception` we can raise here? Raising generic `Exception` would force catching by generic `Exception`, which is we try to avoid